### PR TITLE
strategy: add marketcap strategy

### PR DESCRIPTION
**Pull request by narumiruna**

### CODE REVIEW

File: pkg/fixedpoint/value_map.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+func (m ValueMap) Sub(n ValueMap) ValueMap {
+	if len(m) != len(n) {
+		panic("unequal length")
+	}
+
+	o := ValueMap{}
+
+	for k, v := range m {
+		o[k] = v.Sub(n[k])
```
Comment (zenixls2) [inline pkg/fixedpoint/value_map.go:46]:
```go
nv, ok := n[k]
if !ok {
    panic("key not exists")
}
```

File: pkg/fixedpoint/value_map.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+func (m ValueMap) Mul(n ValueMap) ValueMap {
+	if len(m) != len(n) {
+		panic("unequal length")
+	}
+
+	o := ValueMap{}
+
+	for k, v := range m {
+		o[k] = v.Mul(n[k])
```
Comment (zenixls2) [inline pkg/fixedpoint/value_map.go:60]: same as line 46

File: pkg/fixedpoint/value_map.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	o := ValueMap{}
+
+	for k, v := range m {
+		o[k] = v.Div(n[k])
+	}
+
+	return o
+}
+
+func (m ValueMap) AddScalar(x Value) ValueMap {
```
Comment (zenixls2) [inline pkg/fixedpoint/value_map.go:80]: I recommend using `Add(interface{})` similar to `types.Add` in `types/indicator.go`.
Users could pass in either a `ValueMap` or a `fixedpoint.Value` or even a `float64` to do the job.

File: pkg/strategy/marketcap/types.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package marketcap
```
Comment (zenixls2) [inline pkg/strategy/marketcap/types.go:1]: move this to types?

File: pkg/fixedpoint/value_map.go (hunk 3 above)
Comment (narumiruna) [reply pkg/fixedpoint/value_map.go]: Ok, maybe we can create another PR to do this.

**Review by c9s — APPROVED** [review]

**Pull request merged**